From b60946721941506dd572151f2515152aa7dd7abb Mon Sep 17 00:00:00 2001
From: Saahil Tomar <quic_saahtoma@quicinc.com>
Date: Fri, 16 Jun 2023 15:47:34 +0530
Subject: [PATCH] phy: qcom-qmp-pcie: Fix Pcie Enumeration

This change corrects the PHY sequence to help phy come out of reset,
configure pipe clock rate to 250MHz and update register offsets in rx table

Change-Id: Ibf25b370d7d7006f43f9cfe1b3230984785c803e
Signed-off-by: Saahil Tomar <quic_saahtoma@quicinc.com>
---
 drivers/phy/qualcomm/phy-qcom-qmp-pcie.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/phy/qualcomm/phy-qcom-qmp-pcie.c b/drivers/phy/qualcomm/phy-qcom-qmp-pcie.c
index 907693be35a5..a745369efb6f 100644
--- a/drivers/phy/qualcomm/phy-qcom-qmp-pcie.c
+++ b/drivers/phy/qualcomm/phy-qcom-qmp-pcie.c
@@ -687,8 +687,8 @@ static const struct qmp_phy_init_tbl ipq9574_pcie_rx_tbl[] = {
 	QMP_PHY_INIT_CFG(QSERDES_V4_RX_UCDR_SO_GAIN, 0x02),
 	QMP_PHY_INIT_CFG(QSERDES_V4_RX_UCDR_SO_SATURATION_AND_ENABLE, 0x7F),
 	QMP_PHY_INIT_CFG(QSERDES_V4_RX_UCDR_PI_CONTROLS, 0x70),
-	QMP_PHY_INIT_CFG(QSERDES_V4_RX_RX_EQU_ADAPTOR_CNTRL1, 0x73),
-	QMP_PHY_INIT_CFG(QSERDES_V4_RX_RX_EQU_ADAPTOR_CNTRL2, 0x80),
+	QMP_PHY_INIT_CFG(QSERDES_V4_RX_RX_EQ_OFFSET_ADAPTOR_CNTRL1, 0x73),
+	QMP_PHY_INIT_CFG(QSERDES_V4_RX_RX_OFFSET_ADAPTOR_CNTRL2, 0x80),
 	QMP_PHY_INIT_CFG(QSERDES_V4_RX_RX_MODE_10_LOW, 0x00),
 	QMP_PHY_INIT_CFG(QSERDES_V4_RX_RX_MODE_10_HIGH, 0x02),
 	QMP_PHY_INIT_CFG(QSERDES_V4_RX_RX_MODE_10_HIGH2, 0xC8),
@@ -1850,8 +1850,11 @@ static const struct qmp_phy_cfg ipq9574_gen3x1_pciephy_cfg = {
 	.num_vregs              = 0,
 	.regs                   = ipq_pciephy_gen3_regs_layout,
 
+	.start_ctrl             = SERDES_START | PCS_START,
 	.pwrdn_ctrl             = SW_PWRDN | REFCLK_DRV_DSBL,
 	.phy_status             = PHYSTATUS,
+
+	.pipe_clock_rate        = 250000000,
 };
 
 static const struct qmp_phy_cfg ipq9574_gen3x2_pciephy_cfg = {
@@ -1877,9 +1880,11 @@ static const struct qmp_phy_cfg ipq9574_gen3x2_pciephy_cfg = {
 	.num_vregs              = 0,
 	.regs                   = ipq_pciephy_gen3_regs_layout,
 
+	.start_ctrl             = SERDES_START | PCS_START,
 	.pwrdn_ctrl             = SW_PWRDN | REFCLK_DRV_DSBL,
 	.phy_status             = PHYSTATUS,
 
+	.pipe_clock_rate        = 250000000,
 };
 
 static const struct qmp_phy_cfg sdm845_qmp_pciephy_cfg = {
-- 
2.34.1

